lib/agent0/agent0/hyperdrive/policies/random_hold.py

```diff
@@ -0,0 +1,313 @@
+"""User strategy that opens random positions with set hold position times."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import TYPE_CHECKING
+
+from fixedpointmath import FixedPoint
+
+from agent0.base.types import Trade
+from agent0.hyperdrive import HyperdriveActionType, HyperdriveMarketAction, TradeResult, TradeStatus
+from agent0.hyperdrive.agent import close_long_trade, close_short_trade
+
+from .random import Random
+
+if TYPE_CHECKING:
+    from ethpy.hyperdrive import HyperdriveReadInterface
+    from ethpy.hyperdrive.state import PoolState
+
+    from agent0.hyperdrive import HyperdriveWallet
+
+
+class RandomHold(Random):
+    """Random agent that opens random positions with random hold position times."""
+
+    @classmethod
+    def description(cls) -> str:
+        """Describe the policy in a user friendly manner that allows newcomers to decide whether to use it.
+
+        Returns
+        -------
+        str
+            A description of the policy.
+        """
+        raw_description = """
+        A random policy that randomly generates positions (i.e., long, short, lp) with a random hold time.
+        """
+        return super().describe(raw_description)
+
+    @dataclass(kw_only=True)
+    class Config(Random.Config):
+        """Custom config arguments for this policy
+
+        Attributes
+        ----------
+        max_open_positions: int
+            The maximum number of open positions
+        min_hold_time: int
+            The minimum hold time in seconds. Defaults to 0
+        min_hold_time: int
+            The minimum hold time in seconds. Defaults to 2 * position_duration.
+        """
+
+        max_open_positions: int = 100
+        min_hold_time: int = 0
+        max_hold_time: int | None = None
+
+    @dataclass
+    class _Position:
+        # The minimum close time for this position.
+        # Note it's not guaranteed this position will be closed, but is guaranteed
+        # not to close if the block time is before min_close_time
+        min_close_time: int
+        action_type: HyperdriveActionType
+        bond_amount: FixedPoint
+        maturity_time: int
+        # Status flags
+        ready_to_close: bool = False
+        txn_sent: bool = False
+
+    def __init__(self, policy_config: Config) -> None:
+        """Initializes the bot
+
+        Arguments
+        ---------
+        policy_config: Config
+            The custom arguments for this policy
+        """
+        # Bookkeeping data structure for keeping track of open positions
+        # TODO using a list for now, but likely should use a different data structure
```

Comment:
I think a list is always going to fast enough here, I'd remove the TODO.  If we had a very very large number of positions open maybe, but on the order of 10's just going through the list and filtering is not bad when n is small for  an O(n) operation

Reply:
maybe could use an ordered list in the future that even that seems overkill.

Reply:
While generally I agree with you that a list will likely be sufficient in the majority of cases, I'd argue that `n` in this case isn't on the order of 10's. Specifically, the total number of open positions possible is `(max_hold_time - min_hold_time)/time_per_block`. If you imagine a case where `max_hold_time` is based on position duration, `n` can be quite large. Hence, I'd argue I'll keep this TODO in.